src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -683,6 +681,9 @@ internal static T[] AllocateUninitializedArray<T>(int length)
                 return new T[length];
             }
 #endif
+            if (length < 0)
```

Comment:
This check is unreachable now (except when `Unsafe.SizeOf<T>() * length` overflows - that is a bug on its own).